### ISSUE TITLE
Also carry out  "exam" audit?

### ISSUE THREAD

**Issue by yarikoptic** [Open-Minds-Lab/mrQA#27]
Not quite sure how to name it in addition to "vertical" and "horizontal" (IMHO an easier to grasp something like "cross-{dimension}" would be better so "cross-site" and "cross-subject" correspondingly; and here "cross-sequence").  To do some consistency analysis under assumptions across different sequences within specific exam.  
E.g. correspondence (of shims, geometry, etc) of fieldmaps to func/dwi in BIDS "style" based on their assignment using `IntendedFor` field in side-car files.   Also a check that all func/dwi do have fieldmaps if typically there is a fieldmap.

**Comment by raamana** [comment]
this is a great suggestion Yarik - something I was thinking about too, like `within-session` (vertical) and `cross-session` (horizontal) etc.

**Comment by raamana** [comment]
yes, it would be great to layout all the assumptions and rules explicitly in a consistent framework. this would require careful thought into design on how it will be implemented, and iterated on with some user testing e.g., to enable them to easily add their own new rules.. hence I am leaning into object-oriented design

**Comment by yarikoptic** [comment]
> this is a great suggestion Yarik - something I was thinking about too, like `within-session` (vertical) and `cross-session` (horizontal) etc.

'within' is ambiguous since could still be different dimensions (sessions, sites, ...). `cross` identifies dimension across which consistency is checked

**Comment by raamana** [comment]
I agree. `cross-sequence` is more precise in that sense! @sinhaharsh and will ideate ways to implement this, keeping in mind we need flexibility and easy customization to enable users to implement and share their own rules
